gradoop-data-integration/src/main/java/org/gradoop/dataintegration/transformation/PropagatePropertyToNeighbor.java

```diff
@@ -0,0 +1,246 @@
+/*
+ * Copyright © 2014 - 2018 Leipzig University (Database Research Group)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.gradoop.dataintegration.transformation;
+
+import org.apache.flink.api.common.functions.CoGroupFunction;
+import org.apache.flink.api.common.functions.FlatMapFunction;
+import org.apache.flink.api.common.functions.JoinFunction;
+import org.apache.flink.api.java.DataSet;
+import org.apache.flink.api.java.tuple.Tuple2;
+import org.apache.flink.util.Collector;
+import org.gradoop.common.model.impl.id.GradoopId;
+import org.gradoop.common.model.impl.pojo.Edge;
+import org.gradoop.common.model.impl.pojo.Vertex;
+import org.gradoop.common.model.impl.properties.PropertyValue;
+import org.gradoop.dataintegration.transformation.functions.SetBasedLabelFilter;
+import org.gradoop.flink.model.api.operators.UnaryGraphToGraphOperator;
+import org.gradoop.flink.model.impl.epgm.LogicalGraph;
+import org.gradoop.flink.model.impl.functions.epgm.Id;
+import org.gradoop.flink.model.impl.functions.epgm.SourceId;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * A property of a vertex is propagated to its neighbors and aggregated in a Property List.
+ */
+public class PropagatePropertyToNeighbor implements UnaryGraphToGraphOperator {
+
+  /**
+   * The label of the vertex the property to propagate is part of.
+   */
+  private final String vertexLabel;
+
+  /**
+   * The property key of the property to propagate.
+   */
+  private final String propertyKey;
+
+  /**
+   * The property key where the PropertyValue list should be stored at the target vertices.
+   */
+  private final String targetVertexPropertyKey;
+
+  /**
+   * Only edges with the inserted labels are used. If all labels  are sufficient use 'null'.
+   */
+  private final Set<String> propagatingEdges;
+
+  /**
+   * Only vertices with the inserted labels will store the propagated values.
+   * If all vertices should do it use 'null'
+   */
+  private final Set<String> targetVertexLabels;
+
+  /**
+   * The constructor for the propagate property transformation. Additionally it is possible to
```

Comment:
Give a hint to the other constructor if the user wants to use the two additional arguments. Describe the behavior that results of the two null values.